### ISSUE TITLE
sp_BlitzCache: Incorrect version detection in Azure SQL DB

### ISSUE THREAD

**Issue by erikdarlingdata** [BrentOzarULTD/SQL-Server-First-Responder-Kit#2210]
**Version of the script**
All of them.

**What is the current behavior?**
Version detection for memory grants and spills breaks because Azure SQL DB is 12 whatever

**If the current behavior is a bug, please provide the steps to reproduce.**
Try the memory grant or spills sort order in Azure SQL DB

**What is the expected behavior?**
Should be able to run those sort orders

**Which versions of SQL Server and which OS are affected by this issue? Did this work in previous versions of our procedures?**
Azure SQL DB

**Comment by BrentOzar** [comment]
Whoa, that's odd - we're not checking version numbers for that - I fixed that back in #2015 earlier this year. Here's the relevant code from the current sp_BlitzCache:

```
DECLARE @VersionShowsSpills BIT;
IF EXISTS(SELECT * FROM sys.all_columns WHERE OBJECT_ID = OBJECT_ID('sys.dm_exec_query_stats') AND name = 'max_spills')
    SET @VersionShowsSpills = 1;
ELSE
    SET @VersionShowsSpills = 0;
```

Were you maybe running an old version, or if it's still happening, do you get results in sys.all_columns for those?

**Comment by BrentOzar** [comment]
Found it - was a bug in the Consultant Toolkit scripts that were also checking to see if the sort order was doable. Fixed it over there. Good catch!
